File: 백준/Silver/16435. 스네이크버드/스네이크버드.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    int numOfFruits;
    int snakebirdLength;

    cin >> numOfFruits >> snakebirdLength;

    vector<int> heightOfFruits;

    for (int i = 0; i < numOfFruits; i++) {
        int height;

        cin >> height;

        heightOfFruits.push_back(height);
    }

    int check = 0;

    while (true) {
        check = 0;
        for (int i = 0; i < heightOfFruits.size(); i++) {
            if (snakebirdLength >= heightOfFruits[i]) {
                snakebirdLength++;
                heightOfFruits.erase(heightOfFruits.begin() + i);
                check = 1;
            }
        }

        if (check == 0) {
            break;
        }
    }


    cout << snakebirdLength;
    
    return 0;
}

File: 백준/Silver/16435. 스네이크버드/스네이크버드.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int numOfFruits;
        int snakebirdLength;

        numOfFruits = sc.nextInt();
        snakebirdLength = sc.nextInt();

        int[] heightOfFruits = new int[numOfFruits];

        for (int i = 0; i < numOfFruits; i++) {
            int height;

            height = sc.nextInt();

            heightOfFruits[i] = height;
        }

        int check = 0;

        while (true) {
            check = 0;
            for (int i = 0; i < numOfFruits; i++) {
                if (snakebirdLength >= heightOfFruits[i] && heightOfFruits[i] != -1) {
                    snakebirdLength++;
                    heightOfFruits[i] = -1;
                    check = 1;
                }
            }

            if (check == 0) {
                break;
            }
        }

        System.out.print(snakebirdLength);

        sc.close();
    }
}
